js/ui/overview.js

```diff
@@ -523,6 +526,41 @@ const Overview = new Lang.Class({
         this.hide();
     },
 
+    toggleWindows: function() {
+        if (this.isDummy)
+            return;
+
+        if (!this.visible) {
+            this.showWindows();
+            return;
+        }
+
+        if (!Main.workspaceMonitor.hasActiveWindows) {
```

Comment:
I put this check here because every possible action that comes after this point relies on having active windows in the workspace:
  1. If you have active (hidden or not) windows and you're not in the windows picker => switch to it
  2. Otherwise, if you have active windows **and none** of them are hidden => switch to the apps grid
  3. In any other case (active windows, but all hidden) => hide the overview

All of 1-3 require active windows (hidden or not) to be available, so it made sense to me to add the check right before (1) to early return there, also providing the place to show the "No apps yet" message to the user.